### PR TITLE
Cleanup the stages in the Jenkins file

### LINKED ISSUE #1626: Cleanup the test stages in the Jenkins file
**Is your feature request related to a problem? Please describe.**
As pointed out by @sareh there is a bit of code duplication going on as part of #1620. For the purpose of fixing bbc/simorgh-infrastructure#398 it was suggested to move the cleanup and refactoring work to after the bug has been fixed. 

This issue proposes the necessary cleaning of the Jenkins file steps. 

**Describe the solution you'd like**
To [quote](https://github.com/bbc/simorgh/pull/1620#discussion_r277744865) @sareh 
> The stage `stage('Test Development')` here as well as that below for the `latest` branch are identical and should remain that way. This should be in a reusable function at the top of this file. This will prevent us from having regressions - where one instance is updated and another is forgotten about.
> 
> Likewise the parts of the `stage('Test Production')` that can be re-used, should be in a shared function.

This issue proposes:
- Refactor the steps used to perform the necessary testing into a separate Groovy functions.
- The functions should sit at the top of the Jenkinsfile (right next to the notify slack)
- The `Test Production` steps should also be refactored out to a separate function, and called directly before packaging script is run.

- [x] Initially labelled with ["Refinement needed"](https://github.com/bbc/simorgh/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by DemonusPC**
Resolves #1626

**Overall change:** This cleans up some of the changes that have been made by: #1620. It abstracts out the common steps into its own separate functions. As a bonus I also removed the initial redundant step. 

**Code changes:**

- Moved the steps running the development tests into its own separate function
- Moved the steps running the production tests into its own separate function
- Removed an unnecessary checkout step -> This step became useless since we are pulling the correct branch in the General SCM step provided by jenkins. 

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] ~Tests added for new features~
- [ ] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

**Comment by alex-magana** [comment]
👍  LGTM

File: Jenkinsfile [hunk 1]
```diff
@@ -117,28 +75,23 @@ pipeline {
...
         stage('Test Production and Zip Production') {
           agent {
             docker {
               image "${nodeImage}"
-              label nodeName
               args '-u root -v /etc/pki:/certs'
             }
           }
           steps {
             // Testing
```
Comment (chris-hinds) [inline Jenkinsfile:93]: can we remove this comment, seems unneeded to me, the method name is descriptive enough

**Review by chris-hinds — APPROVED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (DemonusPC) [reply Jenkinsfile:93]: You pointed at the `label nodeName`? Do you want me to remove this?

Comment (chris-hinds) [reply Jenkinsfile:93]: ohhhh no not sure what happened here. just ignore me for now. This can go ahead

**Review by alex-magana — APPROVED** [review]

**Comment by DemonusPC** [comment]
This will be merged on Monday 20th of May

**Pull request merged**